aptos-move/aptos-vm/src/verifier/resource_groups.rs

```diff
@@ -78,37 +91,39 @@ pub(crate) fn validate_resource_groups(
 /// * Verify all changes are compatible upgrades
 /// * Return any new members to validate correctness and all groups to assist in validation
 pub(crate) fn validate_module_and_extract_new_entries(
-    module_storage: &impl AptosModuleStorage,
-    module: &CompiledModule,
-    safer_resource_groups: bool,
+    module_storage: &impl ModuleStorage,
+    new_module: &CompiledModule,
+    features: &Features,
 ) -> VMResult<(
     BTreeMap<String, ResourceGroupScope>,
     BTreeMap<String, StructTag>,
 )> {
     let (new_groups, mut new_members) =
-        if let Some(metadata) = get_metadata_from_compiled_code(module) {
+        if let Some(metadata) = get_metadata_from_compiled_code(new_module) {
             extract_resource_group_metadata(&metadata)?
         } else {
             (BTreeMap::new(), BTreeMap::new())
         };
 
+    let old_module_if_exists =
```

Comment:
nit: we could call this maybe_old_module, maybe_ usually indicates an option.

Reply:
Renamed